tests/loops/test_loops.py

```diff
@@ -611,7 +611,7 @@ def configure_optimizers_multiple(self):
         "epoch_loop.batch_loop.manual_loop.state_dict": ANY,
         "epoch_loop.batch_loop.optimizer_loop.state_dict": {},
         "epoch_loop.batch_loop.optimizer_loop.optim_progress": {
-            "optimizer_idx": n_optimizers,
+            "optimizer_idx": n_optimizers - 1,
```

Comment:
this was because previously, we did +=1 the index at the end of `advance()` but not anymore now